shardy/dialect/sdy/ir/attrs.td

```diff
@@ -345,8 +350,8 @@ def Sdy_DimensionSharding : AttrDef<Sdy_Dialect, "DimensionSharding"> {
 
   let parameters = (ins
     Sdy_AxisRefs:$axes,
-    "bool":$is_closed,
-    OptionalParameter<"std::optional<int64_t>">:$priority
+    AttrOrTypeParameter<"bool", "whether this dimension is closed">:$is_closed,
+    OptionalParameter<"std::optional<int64_t>", "priority of this dimension sharding">:$priority
```

Comment:
hyper nit: These fields live in the dimension sharding, so maybe we don't need to repeat it? AKA "this dimension" and "this dimension sharding". You know better when it comes to better documentation so feel free to reject my suggestion!

Reply:
I see what you mean, I just didn't know how to word this otherwise - would you say something like "priority of sharding" instead?

Reply:
Ahh maybe just "the user priority"? Want to also avoid describing what a priority is in the little docstring here.

Reply:
Ahh maybe even better sorry: "the user priority used during user priority based propagation"

Reply:
I also find "the dimension" or just "dimension sharding" slightly weird, I would go back to "this dimension" or "this dimension sharding" as it sounds more inline with how arguments of a method are documented.

@melissawm @bartchr808 wdyt?

Reply:
> the user priority used during user priority based propagation
SGTM, lets avoid saying user twice, we can drop the first one

Reply:
Tried a slight rewording, let's see if this works.

Reply:
Discussed a bit about this offline, ya happy to revert! While does repeat the attr the property lives in, it can make the description easier to read when it refers back to the attr.